protocol_rfcs/materialize-partition-columns.md

```diff
@@ -0,0 +1,37 @@
+# Materialize Partition Columns
+
+**Associated Github issue for discussions: https://github.com/delta-io/delta/issues/5555**
+
+## Overview
+
+Currently, Delta tables store partition column values primarily in the table metadata (specifically in the `partitionValues` field of `AddFile` actions), and by default these columns are not physically written into the Parquet data files themselves.
+
+This RFC proposes a new writer-only table feature called `materializePartitionColumns`. When enabled, this feature requires partition columns to be physically materialized in Parquet data files alongside the data columns.
+
+## Motivation
+
+This feature provides a mechanism to require partition column materialization at the protocol level, ensuring all writers to the table comply with this requirement during the period when the feature is enabled.
+
+Materializing partition columns enhances compatibility with Parquet readers that access Parquet files directly and do not interpret Delta’s AddFile metadata, as well as with Iceberg readers, which expect partition columns to be stored within the data files.
+
+Additionally, having partition information embedded in the data files themselves enables more flexible data reorganization strategies. The same parquet files could be linked in future versions of a table that do not have the same (or any) partition columns.
+
+--------
+
+
+> ***New Section after Identity Columns section***
+## Materialize Partition Columns
+
+When this feature is enabled, partition columns are physically written to Parquet files alongside the data columns, which can improve flexibility with respect to data layout changes in the future, and make these data files easier to interpret for readers unfamiliar with partition values. To support this feature:
```

Comment:
You can delete `, which can improve flexibility with respect to data layout changes in the future, and make these data files easier to interpret for readers unfamiliar with partition values.`

This RFC is the justification -- we don't need to embed more of that in the protocol spec itself

This RFC should just define what the spec is and what it does/requires. That's all :)